framework/Optimizers/SimulatedAnnealing.py

```diff
@@ -0,0 +1,718 @@
+        @ Out, new, set, modified set of acceptable variables with all formatting complete
+      """
+      # remaking the list is easier than using the existing one
+      acceptable = RavenSampled._formatSolutionExportVariableNames(self, acceptable)
+      new = []
+      print('DEBUGG acceptable:', acceptable)
+      while acceptable:
+        template = acceptable.pop()
+        if '{CONV}' in template:
+          new.extend([template.format(CONV=conv) for conv in self._convergenceCriteria])
+        elif '{VAR}' in template:
+          new.extend([template.format(VAR=var) for var in self.toBeSampled.keys()])
+        else:
+          new.append(template)
+      return set(new)
+
+  def _rejectOptPoint(self, traj, info, old):
+    """
+      Having rejected the suggested opt point, take actions so we can move forward
+      @ In, traj, int, identifier
+      @ In, info, dict, meta information about the opt point
+      @ In, old, dict, previous optimal point (to resubmit)
+    """
+    self._cancelAssociatedJobs(info['traj'], step=info['step'])
+    ## what do do if a point is rejected?
+    # TODO user option to EITHER rerun opt point OR cut step!
+    # initialize a new step
+    self._initializeStep(traj)
+    #self._acceptRerun[traj] = True #TODO: What if the model is inherintly stochastic ?!
+  # END resolving potential opt points
+  # * * * * * * * * * * * * * * * *
+
+  def _applyFunctionalConstraints(self, suggested, previous):
+    """
+      fixes functional constraints of variables in "point" -> DENORMED point expected!
+      @ In, suggested, dict, potential point to apply constraints to
+      @ In, previous, dict, previous opt point in consideration
+      @ Out, point, dict, adjusted variables
+      @ Out, modded, bool, whether point was modified or not
+    """
+    # assume no modifications until proved otherwise
+    modded = False
+    # are we violating functional constraints?
+    passFuncs = self._checkFunctionalConstraints(self.denormalizeData(suggested))
+    # while in violation of constraints ...
+    tries = 500
+    while not passFuncs:
+      modded = True
+      #  try to find new acceptable point
+      denormed = self.denormalizeData(suggested)
+      suggested, _= self._fixFuncConstraintViolations(suggested, previous) # need info, traj
+      denormed = self.denormalizeData(suggested)
+      self.raiseADebug(' ... suggested new opt {}'.format(denormed))
+      passFuncs = self._checkFunctionalConstraints(denormed)
+      tries -= 1
+      if tries == 0:
+        self.raiseAnError(NotImplementedError, 'No acceptable point findable! Now what?')
+    return suggested, modded
+  ###########
+  # Utility Methods #
+  ###########
+  def _temperature(self, fraction):
+    #return max(0.01,min(1,1-fraction))
+    return max(0.01,1-fraction)
+
+  def _coolingSchedule(self, iter, T0):
+    """TODO write the math here"""
+    #defaults
+    if T0 == None:
+      T0 = 1e4
```

Comment:
as said before, just at the "input reading" stage this should be set .

Reply:
removed